src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerMaintainLinkCase.java

```diff
@@ -1,83 +1,110 @@
     private boolean linkCaseEnabled;
 
-    private static final CcdPageConfiguration linkCaseSelectCase = new LinkCaseSelectCase();
+    @Autowired
+    CaseUnlinkedNotification caseUnlinkedNotification;
 
     @Override
     public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
         if (linkCaseEnabled) {
-            doConfigure(configBuilder);
+            new PageBuilder(configBuilder
+                .event(CASEWORKER_MAINTAIN_LINK_CASE)
+                .forStates(Submitted, CaseManagement, AwaitingHearing, AwaitingOutcome)
+                .name("Manage case links")
+                .submittedCallback(this::linkUpdated)
+                .description("To maintain linked cases")
+                .grant(CREATE_READ_UPDATE, SUPER_USER,
+                    ST_CIC_CASEWORKER, ST_CIC_SENIOR_CASEWORKER, ST_CIC_HEARING_CENTRE_ADMIN,
+                    ST_CIC_HEARING_CENTRE_TEAM_LEADER, CASEWORKER_ADMIN_PROFILE)
+                .grantHistoryOnly(
+                    ST_CIC_CASEWORKER,
+                    ST_CIC_SENIOR_CASEWORKER,
+                    ST_CIC_HEARING_CENTRE_ADMIN,
+                    ST_CIC_HEARING_CENTRE_TEAM_LEADER,
+                    ST_CIC_SENIOR_JUDGE,
+                    SUPER_USER,
+                    ST_CIC_JUDGE))
+                .page("maintainCaseLink")
+                .pageLabel("Maintain Case Link")
+                .optional(CaseData::getCaseLinks, ALWAYS_HIDE, null, true)
+                .optional(CaseData::getLinkedCasesComponentLauncher,
+                    null, null, null, null, "#ARGUMENT(UPDATE,LinkedCases)");
         }
     }
 
-    private void doConfigure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
-        PageBuilder pageBuilder = new PageBuilder(configBuilder
-            .event(CASEWORKER_MAINTAIN_LINK_CASE)
-            .forStates(Submitted, CaseManagement, AwaitingHearing, AwaitingOutcome)
-            .name("Links: Maintain Link case")
-            .showSummary()
-            .description("Links: Maintain Link case")
-            .aboutToSubmitCallback(this::aboutToSubmit)
-            .submittedCallback(this::submitted)
-            .grant(CREATE_READ_UPDATE,
-                ST_CIC_CASEWORKER, ST_CIC_SENIOR_CASEWORKER, ST_CIC_HEARING_CENTRE_ADMIN,
-                ST_CIC_HEARING_CENTRE_TEAM_LEADER));
-
-        linkCaseSelectCase.addTo(pageBuilder);
-    }
-
-    @SneakyThrows
-    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
-                                                                       CaseDetails<CaseData, State> beforeDetails) {
-        log.info("Caseworker link the case callback invoked for Case Id: {}", details.getId());
+    public SubmittedCallbackResponse linkUpdated(CaseDetails<CaseData, State> details,
+                                                 CaseDetails<CaseData, State> beforeDetails) {
         var data = details.getData();
```

Comment:
Let's not use var. Needs to be replaced with CaseData.

Reply:
Changed to CaseData